packages/swing-store/src/transcriptStore.js

```diff
@@ -679,19 +727,36 @@ export function makeTranscriptStore(
     }
   }
 
-  function assertComplete(level) {
-    assert.equal(level, 'operational'); // for now
-    // every 'isCurrent' transcript span must have all items
+  function assertComplete(artifactMode) {
+    if (artifactMode === 'debug') {
+      return; // no validation in this mode
+    }
     for (const rec of sqlGetCurrentSpanMetadata.iterate()) {
       const { vatID, startPos, endPos, incarnation } = rec;
-      const count = sqlCountPopulatedSpanItems.get(
-        vatID,
-        incarnation,
-        startPos,
-        endPos,
-      );
-      if (count !== endPos - startPos) {
-        throw Fail`incomplete current transcript span: ${count} items, ${rec}`;
+
+      if (artifactMode === 'operational') {
+        // at 'operational', every 'isCurrent' transcript span must
+        // have all items
+        const count = sqlCountSpanItems.get(vatID, startPos, endPos);
+        if (count !== endPos - startPos) {
+          throw Fail`incomplete current transcript span: ${count} items, ${rec}`;
+        }
+      } else if (artifactMode === 'replay') {
+        // at 'replay', every vat's current incarnation must be fully
+        // populated (which implies 'operational')
+        const incStart = sqlGetStartOfIncarnation.get(vatID, incarnation);
+        const incCount = sqlCountSpanItems.get(vatID, incStart, endPos);
+        if (incCount !== endPos - incStart) {
+          throw Fail`incomplete current incarnation transcript: ${incCount} items`;
+        }
+      } else if (artifactMode === 'archival') {
+        // at 'archival', every incarnation must be fully populated,
+        // which means position=0 up through endPos-1 (which implies
+        // 'replay')
+        const arcCount = sqlCountSpanItems.get(vatID, 0, endPos);
+        if (arcCount !== endPos) {
+          throw Fail`incomplete archival transcript: ${arcCount} vs ${endPos}`;
+        }
       }
```

Comment:
Let's make this throw if we're encountering any other values than operational/replay/archival mode. In particular let's throw if a debug mode is requested (aka remove the return at the top)